### PR TITLE
[4054] Command button visibility

### PR DESCRIPTION
**Pull request by TGazica**
### 🎯 Goal

Hide the commands button if no commands are available.

### 🛠 Implementation details

Added `hasCommands` field to `MessageComposerState` and `MessageInoutView` which determines if the commands button should show inside message composers. 

### 🎨 UI Changes

| Before | After |
| --- | --- |
| Screenshot_1660817785 | Screenshot_1660818375 |

### 🧪 Testing

<details>

<summary>MessageComposer patch</summary>

```
Index: stream-chat-android-ui-common/src/main/kotlin/io/getstream/chat/android/common/composer/MessageComposerController.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/stream-chat-android-ui-common/src/main/kotlin/io/getstream/chat/android/common/composer/MessageComposerController.kt b/stream-chat-android-ui-common/src/main/kotlin/io/getstream/chat/android/common/composer/MessageComposerController.kt
--- a/stream-chat-android-ui-common/src/main/kotlin/io/getstream/chat/android/common/composer/MessageComposerController.kt	(revision 7584ad97ab6617f14b9b7a1da1dd0dbfeadcc06b)
+++ b/stream-chat-android-ui-common/src/main/kotlin/io/getstream/chat/android/common/composer/MessageComposerController.kt	(date 1660817822797)
@@ -288,7 +288,7 @@
     init {
         channelState.flatMapLatest { it.channelConfig }.onEach {
             maxMessageLength = it.maxMessageLength
-            commands = it.commands
+            commands = emptyList()
             state.value = state.value.copy(hasCommands = commands.isNotEmpty())
         }.launchIn(scope)
 

```

</details>

<details>

<summary>MessageInputView patch</summary>

```
Index: stream-chat-android-ui-components/src/main/kotlin/com/getstream/sdk/chat/viewmodel/MessageInputViewModel.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/stream-chat-android-ui-components/src/main/kotlin/com/getstream/sdk/chat/viewmodel/MessageInputViewModel.kt b/stream-chat-android-ui-components/src/main/kotlin/com/getstream/sdk/chat/viewmodel/MessageInputViewModel.kt
--- a/stream-chat-android-ui-components/src/main/kotlin/com/getstream/sdk/chat/viewmodel/MessageInputViewModel.kt	(revision 7584ad97ab6617f14b9b7a1da1dd0dbfeadcc06b)
+++ b/stream-chat-android-ui-components/src/main/kotlin/com/getstream/sdk/chat/viewmodel/MessageInputViewModel.kt	(date 1660818074876)
@@ -22,6 +22,7 @@
 import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.asLiveData
+import androidx.lifecycle.liveData
 import androidx.lifecycle.viewModelScope
 import com.getstream.sdk.chat.utils.extensions.isDirectMessaging
 import com.getstream.sdk.chat.utils.extensions.isModerationFailed
@@ -84,10 +85,10 @@
     /**
      * List of available commands.
      */
-    public val commands: LiveData<List<Command>> =
-        channelState.flatMapLatest { it.channelConfig }.map { config ->
-            config.commands
-        }.asLiveData()
+    public val commands: LiveData<List<Command>> = liveData { emit(emptyList()) }
+        // channelState.flatMapLatest { it.channelConfig }.map { config ->
+        //     config.commands
+        // }.asLiveData()
 
     /**
      * The cooldown interval for the given channel.
Index: stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/application/App.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/application/App.kt b/stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/application/App.kt
--- a/stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/application/App.kt	(revision 7584ad97ab6617f14b9b7a1da1dd0dbfeadcc06b)
+++ b/stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/application/App.kt	(date 1660818074880)
@@ -50,7 +50,7 @@
     private fun initializeToggleService() {
         ToggleService.init(
             applicationContext,
-            mapOf(ToggleService.TOGGLE_KEY_MESSAGE_COMPOSER to true),
+            mapOf(ToggleService.TOGGLE_KEY_MESSAGE_COMPOSER to false),
         )
     }
 

```

</details>

1. Run the app on this branch, the command button should be visible
2. Apply the MessageComposer patch, simulates an empty commands list, the button should not be visible for neither of the MessageComposer or MessageComposerView
3. Apply the second patch and reinstall the app
4. The command button should not be visible for MessageInputView

#### General
- [x] Assigned a person / code owner group (required)
- [x] Thread with the PR link started in a respective Slack channel (#android-chat-core or #android-chat-ui) (required)
- [x] PR targets the `develop` branch
- [x] PR is linked to the GitHub issue it resolves

#### Code & documentation
- [x] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [x] Comparison screenshots added for visual changes
- [x] Affected documentation updated (KDocs, docusaurus, tutorial)

### ☑️Reviewer Checklist
- [ ] UI Components sample runs & works
- [ ] Compose sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [ ] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [ ] All code we touched has new or updated KDocs

### 🎉 GIF

giphy (1)

### CODE REVIEW

File: stream-chat-android-ui-common/src/main/kotlin/io/getstream/chat/android/common/composer/MessageComposerState.kt [hunk 1]
```diff
@@ -48,5 +49,6 @@ public data class MessageComposerState(
     val coolDownTime: Int = 0,
     val messageMode: MessageMode = MessageMode.Normal,
     val alsoSendToChannel: Boolean = false,
-    val ownCapabilities: Set<String> = setOf()
+    val ownCapabilities: Set<String> = setOf(),
+    val hasCommands: Boolean = false
```
Comment (JcMinarro) [inline stream-chat-android-ui-common/src/main/kotlin/io/getstream/chat/android/common/composer/MessageComposerState.kt:53]:
```suggestion
    val hasCommands: Boolean = false,
```

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/input/MessageInputView.kt [hunk 2]
```diff
@@ -296,7 +297,9 @@ public class MessageInputView : ConstraintLayout {
     }
 
     public fun setCommands(commands: List<Command>) {
+        println("commands: ${commands.map { it.name }}")
```
Comment (bychkovdmitry) [inline stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/input/MessageInputView.kt:300]: println?

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/input/MessageInputView.kt [hunk 3]
```diff
@@ -717,7 +720,7 @@ public class MessageInputView : ConstraintLayout {
         binding.sendMessageButtonDisabled.setImageDrawable(drawable)
     }
 
-    private fun configLightningButton() {
+    private fun configCommandsButton() {
```
Comment (bychkovdmitry) [inline stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/input/MessageInputView.kt:723]: Good, less misleading

**Review by bychkovdmitry — APPROVED** [review]
Looks good, tested by clearing commands in the dashboard.

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/input/MessageInputView.kt (hunk 2 above)
Comment (TGazica) [reply stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/input/MessageInputView.kt:300]: just a quick debug, will remove it.

**Pull request merged**